dotcom-rendering/src/web/components/ShowMore.importable.tsx

```diff
@@ -0,0 +1,200 @@
+import { css } from '@emotion/react';
+import {
+	from,
+	neutral,
+	space,
+	visuallyHidden,
+} from '@guardian/source-foundations';
+import {
+	Button,
+	InlineError,
+	SvgCross,
+	SvgPlus,
+} from '@guardian/source-react-components';
+import { useState } from 'react';
+import type { DCRContainerPalette, FEFrontCard } from 'src/types/front';
+import { enhanceCards } from '../../model/enhanceCards';
+import { shouldPadWrappableRows } from '../lib/dynamicSlices';
+import { useApi } from '../lib/useApi';
+import { useOnce } from '../lib/useOnce';
+import { LI } from './Card/components/LI';
+import { UL } from './Card/components/UL';
+import { FrontCard } from './FrontCard';
+
+function decideButtonText({
+	isOpen,
+	loading,
+	containerTitle,
+}: {
+	isOpen: boolean;
+	loading: boolean;
+	containerTitle: string;
+}) {
+	if (isOpen && loading) return 'Loading';
+	if (isOpen) return `Less ${containerTitle}`;
+	return `More ${containerTitle}`;
+}
+
+type Props = {
+	containerTitle: string;
+	path: string;
+	containerId: string;
+	showAge: boolean;
+	containerPalette?: DCRContainerPalette;
+};
+
+export const ShowMore = ({
+	containerTitle,
+	path,
+	containerId,
+	showAge,
+	containerPalette,
+}: Props) => {
+	function findCardLinks() {
+		const containerNode = document.getElementById(containerId);
+		const containerLinksArray = Array.from(
+			containerNode?.querySelectorAll('a') ?? [],
+		);
+		return containerLinksArray.map(
+			(Element) => Element.attributes.getNamedItem('href')?.value,
+		);
+	}
+
+	const [cardLinks, setCardLinks] = useState<string[]>([]);
+
+	useOnce(() => {
+		const linksArray: string[] = findCardLinks().filter(
+			(item): item is string => !!item,
+		);
+		setCardLinks(linksArray);
+	}, [containerId]);
+
+	const [isOpen, setIsOpen] = useState(false);
+
+	function toggleOpen() {
+		setIsOpen(!isOpen);
+	}
+
+	/** We only pass an actual URL to SWR when 'showMore' is true.
+	 * Toggling 'showMore' will trigger a re-render
+	 *   @see https://swr.vercel.app/docs/conditional-fetching#conditional
+	 */
+	const url = isOpen
+		? `https://api.nextgen.guardianapps.co.uk/${path}/show-more/${containerId}.json?dcr=true`
+		: undefined;
+	const { data, error, loading } = useApi<FEFrontCard[]>(url);
+
+	const filteredData =
+		data &&
+		enhanceCards(data).filter((card) => !cardLinks.includes(card.url));
+
+	/**
+		@todo: Desired focus behaviour on expand/collapse?
```

Comment:
there's an issue open for this todo. The rest seem like they can be tackled optionally/at a later date, too.

The content shifting behaviour is something that Frontend seems to have fixed, but which we aren't sure how to address..